File: benchmarks/network/http_server.js
// Node.js HTTP Server for Benchmarking
// Usage: node http_server.js [port]

const http = require('http');

const PORT = process.argv[2] || 8080;
const RESPONSE = 'Hello from Node.js!';

const server = http.createServer((req, res) => {
  res.writeHead(200, {
    'Content-Type': 'text/plain',
    'Content-Length': Buffer.byteLength(RESPONSE)
  });
  res.end(RESPONSE);
});

server.listen(PORT, '0.0.0.0', () => {
  console.log(`Node.js server listening on port ${PORT}`);
});

File: benchmarks/network/http_server.go
// Go HTTP Server for Benchmarking
// Usage: go run http_server.go [port]

package main

import (
	"fmt"
	"net/http"
	"os"
)

const response = "Hello from Go!"

func handler(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Content-Type", "text/plain")
	w.WriteHeader(http.StatusOK)
	w.Write([]byte(response))
}

func main() {
	port := "8080"
	if len(os.Args) > 1 {
		port = os.Args[1]
	}

	http.HandleFunc("/", handler)

	fmt.Printf("Go server listening on port %s\n", port)
	err := http.ListenAndServe(":"+port, nil)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
		os.Exit(1)
	}
}
